core/domain/question_services.py

```diff
@@ -217,3 +228,265 @@ def update_question(
+            given user.
+        question_rights: QuestionRights or None. Rights object for the
+            given question.
+
+    Returns:
+        bool. Whether the given user can edit the given question.
+    """
+    if question_rights is None:
+        return False
+    if role_services.ACTION_EDIT_ANY_QUESTION in user.actions:
+        return True
+
+    return False
+
+
+def assign_role(committer, assignee, new_role, question_id):
+    """Assigns a new role to the user.
+
+    Args:
+        committer: UserActionsInfo. UserActionsInfo object for the user
+            who is performing the action.
+        assignee: UserActionsInfo. UserActionsInfo object for the user
+            whose role is being changed.
+        new_role: str. The name of the new role. Possible values are:
+            ROLE_MANAGER
+        question_id: str. ID of the question.
+
+    Raises:
+        Exception. The committer does not have rights to modify a role.
+        Exception. The assignee is already a manager for the question.
+        Exception. The assignee doesn't have enough rights to become a manager.
+        Exception. The role is invalid.
+    """
+    committer_id = committer.user_id
+    question_rights = get_question_rights(question_id)
+    if (role_services.ACTION_MODIFY_ROLES_FOR_ANY_ACTIVITY not in
+            committer.actions):
+        logging.error(
+            'User %s tried to allow user %s to be a %s of question %s '
+            'but was refused permission.' % (
+                committer_id, assignee.user_id, new_role, question_id))
+        raise Exception(
+            'UnauthorizedUserException: Could not assign new role.')
+
+    assignee_username = user_services.get_username(assignee.user_id)
+    if role_services.ACTION_EDIT_OWNED_QUESTION not in assignee.actions:
+        raise Exception(
+            'The assignee doesn\'t have enough rights to become a manager.')
+
+    old_role = question_domain.ROLE_NONE
+    if question_rights.is_manager(assignee.user_id):
+        old_role = question_domain.ROLE_MANAGER
+
+    if new_role == question_domain.ROLE_MANAGER:
+        if question_rights.is_manager(assignee.user_id):
+            raise Exception('This user already is a manager for this question')
+        question_rights.manager_ids.append(assignee.user_id)
+    elif new_role == question_domain.ROLE_NONE:
+        if question_rights.is_manager(assignee.user_id):
+            question_rights.manager_ids.remove(assignee.user_id)
+        else:
+            old_role = question_domain.ROLE_NONE
+    else:
+        raise Exception('Invalid role: %s' % new_role)
+
+    commit_message = 'Changed role of %s from %s to %s' % (
+        assignee_username, old_role, new_role)
+    commit_cmds = [question_domain.QuestionRightsChange({
+        'cmd': question_domain.CMD_CHANGE_ROLE,
+        'assignee_id': assignee.user_id,
+        'old_role': old_role,
+        'new_role': new_role
+    }).to_dict()]
```

Comment:
No no, let it be the change object here, and convert it to dict just before committing in save_topic_rights. Basically, save_topic_rights should get a list of change objects.

Reply:
Okay done.